src/commons/api/history.rs

```diff
@@ -401,102 +400,173 @@ impl Default for CommandHistoryCriteria {
 #[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
 #[allow(clippy::large_enum_variant)]
 #[serde(rename_all = "snake_case")]
+#[serde(tag = "type")]
 pub enum StorableCaCommand {
     MakeTrustAnchor,
-    ChildAdd(ChildHandle, Option<String>, ResourceSet),
-    ChildUpdateResources(ChildHandle, ResourceSet),
-    ChildUpdateId(ChildHandle, String),
-    ChildCertify(ChildHandle, ResourceClassName, RequestResourceLimit, KeyIdentifier),
-    ChildRevokeKey(ChildHandle, RevocationRequest),
-    ChildRemove(ChildHandle),
+    ChildAdd {
+        child: ChildHandle,
+        ski: Option<String>,
+        resources: ResourceSet,
+    },
+    ChildUpdateResources {
+        child: ChildHandle,
+        resources: ResourceSet,
+    },
+    ChildUpdateId {
+        child: ChildHandle,
+        ski: String,
+    },
+    ChildCertify {
+        child: ChildHandle,
+        resource_class_name: ResourceClassName,
+        limit: RequestResourceLimit,
+        ki: KeyIdentifier,
+    },
+    ChildRevokeKey {
+        child: ChildHandle,
+        revoke_req: RevocationRequest,
+    },
+    ChildRemove {
+        child: ChildHandle,
+    },
     GenerateNewIdKey,
-    AddParent(ParentHandle, StorableParentContact),
-    UpdateParentContact(ParentHandle, StorableParentContact),
-    RemoveParent(ParentHandle),
-    UpdateResourceClasses(ParentHandle, BTreeMap<ResourceClassName, ResourceSet>),
-    UpdateRcvdCert(ResourceClassName, ResourceSet),
-    KeyRollInitiate(i64),
-    KeyRollActivate(i64),
-    KeyRollFinish(ResourceClassName),
-    RoaDefinitionUpdates(RoaDefinitionUpdates),
+    AddParent {
+        parent: ParentHandle,
+        contact: StorableParentContact,
+    },
+    UpdateParentContact {
+        parent: ParentHandle,
+        contact: StorableParentContact,
+    },
+    RemoveParent {
+        parent: ParentHandle,
+    },
+    UpdateResourceEntitlements {
+        parent: ParentHandle,
+        entitlements: Vec<StorableRcEntitlement>,
```

Comment:
Is this change from BTreeMap to Vec also about predictable ordering for testing? Is there any consequence, if possible, of now being able to put the same entitlement into the vec twice which presumably wasn't possible with the BTreeMap?